### ISSUE TITLE
loading of the Retrace configuration via an environment variable

### ISSUE THREAD

**Issue by erikbor** [riboseinc/retrace#334]
Hi @jjr840430 can you implement loading of the Retrace configuration via an environment variable?

Now we can specify a configuration file that can be defined by the environment variable `RETRACE_CONFIG`. Retrace will then open that file and get its configuration.

To make mass fuzzing faster it would be beneficial if we can have the entire configuration in a special environment variable.

This is an example of how we could put the configuration in the environment:
```C
#include <stdio.h>
#include <stdlib.h>

int
main()
{
    char retraceconf[512];

    snprintf(retraceconf, sizeof(retraceconf), "logging-global,LOG_GROUP_NET,LOG_LEVEL_ALL\nstringinject,sendto,chr(\\ff),10:8,1");
    setenv("RETRACE_CONFIG", retraceconf, 1);
    printf("%s\n", getenv("RETRACE_CONFIG"));
    return(0);
}
```

**Comment by jjr840430** [comment]
@erikbor OK, I will implement it. Thanks.

**Comment by jjr840430** [comment]
@erikbor `RETRACE_CONFIG` is already used as an environment variable for specifying the configuration file path.

What about using `RETRACE_CONFIG_VAR` instead it.

**Comment by erikbor** [comment]
Any variable name is fine, so `RETRACE_CONFIG_VAR` is alright. Thanks!

**Comment by jjr840430** [comment]
By the way, is it possible to modify configuration var while tracing?
I think it's possible case when tracing `daemon` processes, so we may update configurations while running `daemon`.

**Comment by erikbor** [comment]
Yeah that will be one of the key modifications that needs to be done soon. I described the first steps in the discussion with you and Ilja. I am writing a more detailed explanation with a diagram and will soon share that for your review.

**Comment by jjr840430** [comment]
Another question, when `retrace` has the configuration file using `-f` option and `RETRACE_CONFIG_VAR` env, then which configuration should be loaded?

**Comment by erikbor** [comment]
Maybe check whether the first line of `RETRACE_CONFIG` results in a successful `stat()` which would mean that's a valid file and then fopen it. If the `stat()` fails then proceed with loading the config from that environment variable?
Do you think that's a good approach?

**Comment by jjr840430** [comment]
I think `RETRACE_CONFIG_VAR` may remain after completed tracing for the specific application.
If another application should be traced, then old `RETRACE_CONFIG_VAR` may bring some conflicts.

I think `RETRACE_CONFIG_VAR` should be ignored if `-f` option is specified.

What do you think about it?

**Comment by erikbor** [comment]
Okay cool

**Comment by jjr840430** [comment]
@erikbor Implemented by #335

**Comment by riataman** [comment]
closed per above comment
